### ISSUE TITLE
BUG SensitivityReader storing data from second to last latent generation

### ISSUE THREAD

**Issue by drewejohnson** [CORE-GATECH-GROUP/serpent-tools#365]
## Summary of issue
Sensitivity files for Serpent 2.1.31 include (by default or with additional options) data from intermediate latent generations. The naming of these variables overwrites the "correct" data, e.g. data for sensitivity of keff is taken from the second to last latent generation rather than all latent generations. This issue was reported [on the discussion forum](https://ttuki.vtt.fi/serpent/viewtopic.php?f=30&t=3278)

## Code for reproducing the issue
New `KEFF_SENS` data are provided as
```
59:ADJ_PERT_KEFF_SENS = [
63:ADJ_PERT_KEFF_SENS = reshape(ADJ_PERT_KEFF_SENS, [2, SENS_N_ENE, SENS_N_PERT, SENS_N_ZAI, SENS_N_MAT]);
64:ADJ_PERT_KEFF_SENS = permute(ADJ_PERT_KEFF_SENS, [5, 4, 3, 2, 1]);
66:ADJ_PERT_KEFF_SENS_E_INT = [
70:ADJ_PERT_KEFF_SENS_E_INT = reshape(ADJ_PERT_KEFF_SENS_E_INT, [2, SENS_N_PERT, SENS_N_ZAI, SENS_N_MAT]);
71:ADJ_PERT_KEFF_SENS_E_INT = permute(ADJ_PERT_KEFF_SENS_E_INT, [4, 3, 2, 1]);
282:ADJ_PERT_KEFF_SENS_0_GEN = [
286:ADJ_PERT_KEFF_SENS_0_GEN = reshape(ADJ_PERT_KEFF_SENS_0_GEN, [2, SENS_N_ENE, SENS_N_PERT, SENS_N_ZAI, SENS_N_MAT]);
287:ADJ_PERT_KEFF_SENS_0_GEN = permute(ADJ_PERT_KEFF_SENS_0_GEN, [5, 4, 3, 2, 1]);
289:ADJ_PERT_KEFF_SENS_0_GEN_E_INT = [
293:ADJ_PERT_KEFF_SENS_0_GEN_E_INT = reshape(ADJ_PERT_KEFF_SENS_0_GEN_E_INT, [2, SENS_N_PERT, SENS_N_ZAI, SENS_N_MAT]);
294:ADJ_PERT_KEFF_SENS_0_GEN_E_INT = permute(ADJ_PERT_KEFF_SENS_0_GEN_E_INT, [4, 3, 2, 1]);
493:ADJ_PERT_KEFF_SENS_1_GEN = [
497:ADJ_PERT_KEFF_SENS_1_GEN = reshape(ADJ_PERT_KEFF_SENS_1_GEN, [2, SENS_N_ENE, SENS_N_PERT, SENS_N_ZAI, SENS_N_MAT]);
...
2181:ADJ_PERT_KEFF_SENS_9_GEN = [
2185:ADJ_PERT_KEFF_SENS_9_GEN = reshape(ADJ_PERT_KEFF_SENS_9_GEN, [2, SENS_N_ENE, SENS_N_PERT, SENS_N_ZAI, SENS_N_MAT]);
2186:ADJ_PERT_KEFF_SENS_9_GEN = permute(ADJ_PERT_KEFF_SENS_9_GEN, [5, 4, 3, 2, 1]);
2188:ADJ_PERT_KEFF_SENS_9_GEN_E_INT = [
2192:ADJ_PERT_KEFF_SENS_9_GEN_E_INT = reshape(ADJ_PERT_KEFF_SENS_9_GEN_E_INT, [2, SENS_N_PERT, SENS_N_ZAI, SENS_N_MAT]);
2193:ADJ_PERT_KEFF_SENS_9_GEN_E_INT = permute(ADJ_PERT_KEFF_SENS_9_GEN_E_INT, [4, 3, 2, 1]);
```

## Actual outcome including console output and error traceback if applicable
The method for processing these arrays 
https://github.com/CORE-GATECH-GROUP/serpent-tools/blob/b1aa25599861b4d75ed84f4fffaa0b12a3871a47/serpentTools/parsers/sensitivity.py#L231-L236
doesn't contain logic for these `N_GEN` names, and then overwrites the useful information, e.g. data from `ADJ_PERT_KEFF_SENS_9_GEN` is stored in place of `ADJ_PERT_KEFF_SENS` as `sensitivities["keff"]`

## Expected outcome
Store data from all latent generations, `ADJ_PERT_KEFF_SENS` as the actual data, with options to store data from latent generations as well.

## Versions

Please provide the following:
Per [the reporting user](https://ttuki.vtt.fi/serpent/viewtopic.php?f=30&t=3278#p10229)
* Version  from ``serpentTools.__version__`` `0.9.0`
* Python version - ``python --version`` N/A
* IPython or Jupyter version if applicable N/A
* If using the development version, the output from ``git describe``

**Comment by drewejohnson** [comment]
Merged into master with #366 and will be a part of 0.9.1 release

**Comment by drewejohnson** [comment]
Merged into develop with #369
